src/main/kotlin/io/cloudflight/gradle/autoconfigure/java/JavaConfigurePlugin.kt

```diff
@@ -24,15 +28,26 @@ private const val TESTNG_ARTIFACT_GROUP = "org.testng"
 
 class JavaConfigurePlugin : Plugin<Project> {
     override fun apply(project: Project) {
+        project.rootProject.plugins.apply(JavaAutoConfigurePlugin::class)
         project.plugins.apply(JavaLibraryPlugin::class)
 
         val gradle = project.gradle
         val extensions = project.extensions
         val tasks = project.tasks
 
-        val javaConfigureExtension = extensions.create(EXTENSION_NAME, JavaConfigurePluginExtension::class)
+        val autoConfigure = project.rootProject.extensions.getByType(AutoConfigureExtension::class.java)
```

Comment:
well, the thing we the current approach is that the `JavaConfigurePlugin` does not know itself about the conventions of its properties, they all come from the `autoConfigure` block and its conventions on the root module.

Actually I'm not yet sure if having that global `autoConfigure` block is a good idea, probably it would be better to let users work with an `allprojects {}` block and in there do something like `if JavaConfigurePlugin applied, then change JavaConfigurePluginExtension`